### PR TITLE
Update readme

### PR DESCRIPTION
**Pull request by NikolaevaIrina**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,42 @@
...
+This example demonstrates how to customize standard authentication behavior and supply additional logon parameters in XAF applications. The default login form displays **User Name** and **Password** editors. This sample customizes the login form to show **Company** and **Application User** lookup editors instead of **User Name**.
 
-Under constuction.
+![Login form](https://docs.devexpress.com/eXpressAppFramework/images/custom-logon-prameters-blazor.png)
+
+## Implementation Details
+
+### ASP.NET Core Blazor
+
+1. Define custom parameters. To do this, add a new class with parameters to your project. In this example, this is the [Company](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/Company.cs) class. This class should contain company names and a list of `ApplicationUser` objects as a part of a one-to-many relationship. Then, add the second part of this relationship to the [ApplicationUser](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/ApplicationUser.cs#L28) class generated by the Solution Wizard. And finally, add the `Company` class to your application’s [DbContext](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/EFCoreCustomLogonAllDbContext.cs#L43).
```
Comment (AndreyKozhevnikov) [inline README.md]: 'Define a custom class that we add to the logon window. In this example this is the Company class that contains company names. This class should contain a list of 'ApplicationUser'.....

File: README.md [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
+2. Add custom parameters to the login window. For an example, refer to the following file - [CustomLogonParameters.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomLogonParameters.cs).
```
Comment (AndreyKozhevnikov) [inline README.md]: "Create a custom logon parameter class that uses the Company class." (on this step we don't add anything to the login window)

File: README.md [hunk 3]
```diff
@@ -1,9 +1,42 @@
...
+3. Implement [custom authentication](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthentication.cs).
```
Comment (AndreyKozhevnikov) [inline README.md]: ... that uses the created custom logon parameters.

File: README.md [hunk 4]
```diff
@@ -1,9 +1,42 @@
...
+4. Implement a custom [authentication provider](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthenticationStandardProvider.cs).
```
Comment (AndreyKozhevnikov) [inline README.md]: ... that creates an instance of the created custom authentication

File: README.md [hunk 5]
```diff
@@ -1,9 +1,42 @@
...
+5. Pass custom classes to the Security System. For an example, refer to the following file - [Startup.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Startup.cs#L73-L78).
+6. Add demo data.
```
Comment (AndreyKozhevnikov) [inline README.md]: link to https://github.com/DevExpress-Examples/xaf-custom-logon-parameters/blob/update-readme/CS/EF/EFCoreCustomLogonAll.Module/DatabaseUpdate/Updater.cs

File: README.md [hunk 6]
```diff
@@ -1,9 +1,42 @@
...
+7. Generate a demo database.
```
Comment (AndreyKozhevnikov) [inline README.md]: link to https://docs.devexpress.com/eXpressAppFramework/404264/data-security-and-safety/security-system/authentication/customize-standard-authentication-behavior-and-supply-additional-logon-parameters/customize-authentication-behavior-blazor#generate-a-demo-database

File: README.md (hunk 1 above)
Comment (NikolaevaIrina) [reply README.md]:
```suggestion
1. Define a custom class to be added to the logon window. In this example, this is the [Company](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/Company.cs) class that contains company names. This class should contain a list of `ApplicationUser` objects as a part of a one-to-many relationship. Then, add the second part of this relationship to the [ApplicationUser](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/ApplicationUser.cs#L28) class generated by the Solution Wizard. And finally, add the `Company` class to your application’s [DbContext](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/EFCoreCustomLogonAllDbContext.cs#L43).
```

File: README.md (hunk 2 above)
Comment (NikolaevaIrina) [reply README.md]:
```suggestion
2. Create a custom logon parameter class that uses the `Company` class. For an example, refer to the following file - [CustomLogonParameters.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomLogonParameters.cs).
```

File: README.md (hunk 3 above)
Comment (NikolaevaIrina) [reply README.md]:
```suggestion
3. Implement [custom authentication](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthentication.cs) that uses created custom logon parameters..
```

File: README.md (hunk 4 above)
Comment (NikolaevaIrina) [reply README.md]:
```suggestion
4. Implement a custom [authentication provider](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthenticationStandardProvider.cs) that creates an instance of the defined custom authentication.
```

File: README.md [hunk 7]
```diff
@@ -1,9 +1,42 @@
...
+2. Create a custom logon parameter class that uses the `Company` class. For an example, refer to the following file - [CustomLogonParameters.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomLogonParameters.cs).
+3. Implement [custom authentication](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthentication.cs) that uses created custom logon parameters..
+4. Implement a custom [authentication provider](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthenticationStandardProvider.cs) that creates an instance of the created custom authentication.
```
Comment (NikolaevaIrina) [inline README.md]:
```suggestion
4. Implement a custom [authentication provider](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthenticationStandardProvider.cs) that creates an instance of the defined custom authentication.
```

File: README.md (hunk 5 above)
Comment (NikolaevaIrina) [reply README.md]:
```suggestion
6. Add [demo data](./CS/EF/EFCoreCustomLogonAll.Module/DatabaseUpdate/Updater.cs).
```

File: README.md (hunk 6 above)
Comment (NikolaevaIrina) [reply README.md]:
```suggestion
7. Generate a [demo database](https://docs.devexpress.com/eXpressAppFramework/404264/data-security-and-safety/security-system/authentication/customize-standard-authentication-behavior-and-supply-additional-logon-parameters/customize-authentication-behavior-blazor#generate-a-demo-database).
```

File: README.md [hunk 8]
```diff
@@ -1,9 +1,42 @@
...
+This example demonstrates how to customize standard authentication behavior and supply additional logon parameters in XAF applications. The default login form displays **User Name** and **Password** editors. This sample customizes the login form to show **Company** and **Application User** lookup editors instead of **User Name**.
 
-Under constuction.
+![Login form](https://docs.devexpress.com/eXpressAppFramework/images/custom-logon-prameters-blazor.png)
+
+## Implementation Details
+
+### ASP.NET Core Blazor
+
+1. Define a custom class to be added to the logon window. In this example, this is the [Company](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/Company.cs) class that contains company names. This class should contain a list of `ApplicationUser` objects as a part of a one-to-many relationship. Then, add the second part of this relationship to the [ApplicationUser](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/ApplicationUser.cs#L28) class generated by the Solution Wizard. And finally, add the `Company` class to your application’s [DbContext](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/EFCoreCustomLogonAllDbContext.cs#L43).
```
Comment (Abadzhev) [inline README.md]: You start with "define a custom class to be added to the logon window". I'm not sure what that means. You already described the task above - "replace User Name with Company and Application User editors". We never said anything about "adding classes to the window". That might seem confusing, I think. I would rather remove that first sentence and go directly to the sentence about the Company class. It should be more straightforward, as we already mentioned that we need a Company editor. 

```suggestion
1. Define a [Company](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/Company.cs) class that contains company names. This class should contain a list of `ApplicationUser` objects as a part of a one-to-many relationship. Add the second part of this relationship to the [ApplicationUser](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/ApplicationUser.cs#L28) class generated by the Solution Wizard. Finally, add the `Company` class to your application’s [DbContext](./CS/EF/EFCoreCustomLogonAll.Module/BusinessObjects/EFCoreCustomLogonAllDbContext.cs#L43).
```

File: README.md [hunk 9]
```diff
@@ -1,9 +1,42 @@
...
+2. Create a custom logon parameter class that uses the `Company` class. For an example, refer to the following file - [CustomLogonParameters.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomLogonParameters.cs).
```
Comment (Abadzhev) [inline README.md]: For an example - looks very similar to "for example", so may be confusing. And it's not something that you would see often. 

```suggestion
2. Create a custom logon parameter class that uses the `Company` class. To see an example, refer to the following file - [CustomLogonParameters.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomLogonParameters.cs).
```

File: README.md [hunk 10]
```diff
@@ -1,9 +1,42 @@
...
+3. Implement [custom authentication](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthentication.cs) that uses created custom logon parameters..
```
Comment (Abadzhev) [inline README.md]:
```suggestion
3. Implement [custom authentication](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthentication.cs) that uses custom logon parameters.
```

File: README.md [hunk 11]
```diff
@@ -1,9 +1,42 @@
...
+4. Implement a custom [authentication provider](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthenticationStandardProvider.cs) that creates an instance of the defined custom authentication.
```
Comment (Abadzhev) [inline README.md]: Similar as in the sentence before. If you already say "custom", it already implies that it's something you "created" or "defined" manually in this example. You don't need to say "created custom" or "defined custom". Such phrases seem redundant. 

 ```suggestion
4. Implement a custom [authentication provider](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Security/CustomAuthenticationStandardProvider.cs) that creates an instance of the custom authentication strategy.
```

File: README.md [hunk 12]
```diff
@@ -1,9 +1,42 @@
...
+5. Pass custom classes to the Security System. For an example, refer to the following file - [Startup.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Startup.cs#L73-L78).
```
Comment (Abadzhev) [inline README.md]:
```suggestion
5. Pass custom classes to the Security System. To see an example, refer to the following file - [Startup.cs](./CS/EF/EFCoreCustomLogonAll.Blazor.Server/Startup.cs#L73-L78).
```

**Review by dirk-pieterse — APPROVED** [review]

**Pull request merged**